### PR TITLE
Support for P5.

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Support protocol 5.

## Changes

- Support for protocol version 5.
- Support for emitted events from smart contract upgrading.
- Various bug fixes (see CHANGELIG.md)

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
 
 ## Unreleased changes
 
+- Support for P5
+- Fix a bug where pending changes for delegators was not visible via `getAccountInfo`.
```
Comment (abizjak) [inline CHANGELOG.md:6]:
```suggestion
- Fix a bug where pending changes for delegators were not visible via `getAccountInfo`.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/updates/queues/TimeParameters.java [hunk 2]
```diff
@@ -19,10 +19,10 @@ public final class TimeParameters {
     /**
      * Mint rate per payday (as a proportion of the extant supply).
      */
-    private final MintRate mintPerPayday;
+    private final double mintPerPayday;
```
Comment (abizjak) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/updates/queues/TimeParameters.java:22]: Well this is not entirely accurate. Why the change?

At least we need to document a bit that there might be rounding issues.

Comment (EmilLa1) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/updates/queues/TimeParameters.java:24]: I have documented the caveat here. 
The reason for the change was a bugfix, as the 'mintPerPayday' is being returned as a number as opposed to a struct with an exponent and mantissa.

**Review by abizjak — APPROVED** [review]
